### PR TITLE
fix(nextjs): Update params type with Promise as expected in Next.js 15

### PR DESCRIPTION
**Pull request by ggallon**
## ☕️ Reasoning

Follow #12002, fix type error with auth() wapprer,  update params type with Promise in handler.
Ref: [Next.js Doc](https://nextjs.org/docs/canary/app/api-reference/file-conventions/route#context-optional)

## 🧢 Checklist

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks! I believe we should use our `Awaitable` here from `@auth/core/types`, to be compatible with Next.js 14.

**Comment by ggallon** [comment]
@balazsorban44, It's done

**Comment by ggallon** [comment]
@balazsorban44 sorry, that's not the way. `pnpm tsc --noEmit` in locale not always cause an error (??) but every time with `pnpm next build`:
```
Type "AppRouteHandlerFnContext" is not a valid type for the function's second argument.
    Expected "Promise<any>", got "Awaitable<Record<string, string | string[]>> | undefined".
      Expected "Promise<any>", got "undefined".
````

**Comment by Aralf** [comment]
> @balazsorban44 sorry, that's not the way. `pnpm tsc --noEmit` in locale not always cause an error (??) but every time with `pnpm next build`:
> 
> ```
> Type "AppRouteHandlerFnContext" is not a valid type for the function's second argument.
>     Expected "Promise<any>", got "Awaitable<Record<string, string | string[]>> | undefined".
>       Expected "Promise<any>", got "undefined".
> ```

@ggallon as the error mentions, `undefined` is not valid, that's because `params` is set as optional, it shouldn't be.

File: packages/next-auth/src/lib/types.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import type { NextRequest } from "next/server"
+import type { Awaitable } from "@auth/core/types"
 
 /**
  * AppRouteHandlerFnContext is the context that is passed to the handler as the
  * second argument.
  */
 export type AppRouteHandlerFnContext = {
-  params?: Record<string, string | string[]>
+  params?: Awaitable<Record<string, string | string[]>>
```
Comment (rpavlovs) [inline packages/next-auth/src/lib/types.ts:10]:
```suggestion
  params: Awaitable<Record<string, string | string[]>>
```

**Comment by rpavlovs** [comment]
@balazsorban44 could you please bump the npm package version?

**Comment by rpavlovs** [comment]
@balazsorban44 Please make an npm push. This is blocking Next 15, React 19 update.

**Pull request merged**